pkg/services/object/acl/v2/service.go

```diff
@@ -491,14 +491,9 @@ func (b Service) PutRequestToInfo(request *protoobject.PutRequest) (RequestInfo,
 		}
 	}
 
-	if !replication {
-		// header length is unchecked for replication because introducing a restriction
-		// should not prevent the replication of objects created before.
-		// See also https://github.com/nspcc-dev/neofs-api/issues/293
-		var hdrLen = header.MarshaledSize()
-		if hdrLen > objectsdk.MaxHeaderLen {
-			return RequestInfo{}, user.ID{}, fmt.Errorf("object header length exceeds the limit: %d>%d", hdrLen, objectsdk.MaxHeaderLen)
-		}
+	var hdrLen = header.MarshaledSize()
+	if hdrLen > objectsdk.MaxHeaderLen {
```

Comment:
Not sure why we're dealing with it here at all, it's supposed to be about permissions, not format correctness.

Reply:
idk, but it cannot be put to the "validator" that we have now, since it is reused for both cases: prepared and unprepared objects, so a header without checksums, etc is not a full header

Reply:
agree, we no longer depend on replicate-ness, so better to move to fmt space

> unprepared objects, so a header without checksums, etc is not a full header

unprepared header can only grow

Reply:
> unprepared header can only grow

if an unprepared header is 16k-1 len, a prepared one overflows for sure

Reply:
This fact is unrelated to the place for this condition